src/java_bytecode/java_bytecode_instrument.cpp

```diff
@@ -566,6 +566,38 @@ void java_bytecode_instrumentt::operator()(exprt &expr)
   instrument_code(expr);
 }
 
+/// Instruments the code attached to `symbol` with
+/// runtime exceptions or corresponding assertions.
+/// Exceptions are thrown when the `throw_runtime_exceptions` flag is set.
+/// Otherwise, assertions are emitted.
+/// \par parameters: `symbol_table`: global symbol table (may gain exception type
+///   stubs and similar)
+/// `symbol`: the symbol to instrument
+/// `throw_runtime_exceptions`: flag determining whether we instrument the code
+/// with runtime exceptions or with assertions. The former applies if this flag
+/// is set to true.
+/// `max_array_length`: maximum array length; the only reason we need this is
+/// in order to be able to call the object factory to create exceptions.
+void java_bytecode_instrument(
```

Comment:
You might consider renaming this function to be `java_bytecode_instrument_symbol` to make its difference clear from the overloaded method.